### PR TITLE
Fixed inconsistent chaplain immunities

### PR DESCRIPTION
**Pull request by PrimeNumb**
## About the PR 
There were some spells and abilities from both vampires and wizards that still worked on the chaplain, despite the chaplain being regarded as the only person immune to supernatural and magical abilities. This PR makes it so the chaplain - in addition to existing immunities - is also protected from:

- Staff of Cthulu direct attacks (trying to pick it up will still result in damage)
- Bull's charge
- Prismatic Spray
- Fireball (yes, the chaplain was in fact not immune to these)
- Frost bats

Do note that Fireball can still do splash damage.

## Why's this needed? 
The chaplain was the only person on the station immune to supernatural and magical abilities, and the previously mentioned inconsistent protection from these abilities was unintuitive; players would have no idea which abilities worked and which abilities didn't outside trial and error.

## Changelog

```
(u)RichardGere:
(+)Fixed chaplain immunity inconsistencies. For a full list of covered abilities, please see the PR.
```

### CODE REVIEW

File: code/obj/item/misc_wizard_stuff.dm [hunk 1]
```diff
@@ -165,8 +165,9 @@
 	desc = "A dark staff infused with eldritch power. Trying to steal this is probably a bad idea."
 	icon_state = "staffcthulhu"
 	item_state = "staffcthulhu"
-	force = 14
+	force = 0
 	hitsound = 'sound/effects/ghost2.ogg'
+	var/magic_damage = 14
```
Comment (PrimeNumb) [inline code/obj/item/misc_wizard_stuff.dm:170]: I only made this a var incase someone wants to ~~commit crimes~~ varedit for whatever reason.

File: code/obj/artifacts/artifactdatums.dm [hunk 2]
```diff
@@ -182,7 +182,16 @@
...
 		..()
 		src.randomise()
+
+	on_hit(atom/hit, angle, var/obj/projectile/O)
+		var/mob/M = hit
+		if (istype(M) && M?.traitHolder?.hasTrait("training_chaplain"))
+			M.visible_message("<span class='alert'>A divine light shields [hit]!</span>")
+			return
+
+		..()
```
Comment (Tarmunora) [inline code/obj/artifacts/artifactdatums.dm:197]: Isn't this handled in `bullet_act()`?

File: code/obj/item/misc_wizard_stuff.dm (hunk 1 above)
Comment (Tarmunora) [reply code/obj/item/misc_wizard_stuff.dm:170]: Note that this makes it unable to say, bash a window. Intentional?

File: code/obj/artifacts/artifactdatums.dm (hunk 2 above)
Comment (PrimeNumb) [reply code/obj/artifacts/artifactdatums.dm]: this was just me being sloppy with cleaning up from a rewrite

File: code/obj/item/misc_wizard_stuff.dm (hunk 1 above)
Comment (PrimeNumb) [reply code/obj/item/misc_wizard_stuff.dm]: It's funny that I never seriously considered someone might actually use this for anything other than bashing someone's head in. I think it's a bit silly that the staff doesn't do strictly "magical" damage and can actually be used to bash other things down, but that's outside the scope of this PR and a discussion that I'd rather not start here. 

Reverted to a previous revision.

**Pull request merged**